Changes to create diagram with graphviz Addresses: #164

GadgetSteve opened the pull request:
Changes to conf.py to:

  1. Enable graphviz extension 
  2. Set graphviz output to SVG format
  3. Add some comments on how to get graphviz working from sphinx on windows

Changes to docs/source/projects/content-projects.rst to replace current, (neater), png with graphviz section that produces .svn image which while not as neat does have working links.

Tested for: html, htmldir, htmlsingle & epub on Windows 10.

Carreau commented on docs/source/conf.py:
```diff
@@ -62,7 +63,7 @@
 # The short X.Y version.
 version = '4.1'
 # The full version, including alpha/beta/rc tags.
-release = '4.1.0'
```
We would need to change that I think.

Carreau commented:
Hi @GadgetSteve thanks a lot for doing that !

Is graphviz available on read the docs ? Otherwise that would break the autobuild. 

@willingc you are the documentation master, if this is fine with you feel free to go ahead with it .

GadgetSteve commented:
@Carreau - A quick check on [Read The Docs](http://read-the-docs.readthedocs.io/en/latest/builds.html#packages-installed-in-the-build-environment) lists graphviz as one of the installed libraries so it **_should_** just "work out of the box".

willingc commented on docs/source/projects/content-projects.rst:
```diff
@@ -13,8 +13,54 @@ Installation
+	digraph decide_project {
+		size="10,4";
+		// Top Level
+		//label="How do I decide which pakages I need?";
```
@GadgetSteve Typo in spelling of `packages` in this line and the next line.

willingc commented on docs/source/projects/content-projects.rst:
```diff
@@ -13,8 +13,54 @@ Installation
+		// 3rd Level
+		online [label="Try Online", href="https://try.jupyter.org/", target="_blank", tooltip="Try Online", style=underline, color=blue,]
+		doinstall [label="Install Jupyter", tooltip="How to install", href="../install.html", target="_top", color=green,] // 4th level
+		hub [label="Jupyter Hub", tooltip="Install Jupyter Hub", target="_blank", color=blue, href="https://github.com/jupyterhub/jupyterhub"]
```
JupyterHub should be one word.

willingc commented on docs/source/projects/content-projects.rst:
```diff
@@ -13,8 +13,54 @@ Installation
+		my [shape=plaintext, label="On my system", tooltip="Local Installation", color=blue]
+		// below kernel
+		widgets [label="Widgets", tooltip="Install & use ipywidgets", target="_blank", color=blue, href="https://ipywidgets.readthedocs.io/en/latest/"]
+		extend [label="Extensions", tooltip="Install & use extentions", target="_blank", color=blue, href="https://github.com/jupyter/help"]
```
Typo in tooltip: should be `extensions`

willingc commented:
@GadgetSteve I'm sorry for the delay in reviewing and thank you for the contribution :smile:  Thanks @Carreau for doing the initial review and following up.

I've left a few comments on the code (mostly typos). I am currently traveling, but in the next couple days, I will check the `href` links a bit more closely and compare them to the links in the original visual.

I appreciate your PR and interest in improving our documentation. I'm looking forward to getting this merged after correcting the typos and links.

GadgetSteve commented:
@willingc & @Carreau - thank you for the kind words:

 - I have corrected the typos noted above _spelling was never my strong point_ 
 - Laid things out in a more readable, pythonic and consistent manner.
 - Added a key to the colours of the link shapes.
 - Added a font selection which I think makes it much more readable.
 - Added missing tool tips to several of the items.

GadgetSteve commented:
Please note that the Travis build system does show one warning due to being unable to find 'dot' on the travis build system - I would guess that adding to .travis.yml, near the beginning, something like:

before_install:
	- sudo apt-get update -qq
	- sudo apt-get install graphviz 

Should do the job but I have very limited experience of travis-ci so have no way of knowing if this is a container based build _if it is then the command syntax will be different - see [here](https://docs.travis-ci.com/user/installing-dependencies/#Installing-Packages-on-Container-Based-Infrastructure)_

willingc commented:
@GadgetSteve I'm happy to merge this PR. Thanks for improving the docs and creating active links. I hope you consider contributing again in the future. :cake: 🎉

GadgetSteve commented:
My pleasure Carol _@willingc_ it also provided me with a useful learning experience of both the dot language and of the sphinx build system. I also found out about some Jupyter features that I was unaware of in the process which is always to the good. Thank-you for your awesome work on the Jupyter project along with all of the devs.
